From be78589fe0289671e2ff18fc95fb545b6b83de6a Mon Sep 17 00:00:00 2001
From: Octavian Purdila <octavian.purdila@intel.com>
Date: Thu, 27 Sep 2012 19:28:43 +0300
Subject: devices: allow sending on device-add notifications for major/minor-less devices

This is usefull for getting device plug/unplug notifications for
things like USB ethernet adapters which do not have major/minor
numbers.

Issue: ADEV-21
Change-Id: Ie11c3da54438d6291bb022f9eef42d93ca7244eb
Category: AOSP Improvement
Domain: AOSP-init
Upstream-candidate: Yes
Signed-off-by: Octavian Purdila <octavian.purdila@intel.com>
---
 init/devices.c |   10 ++++------
 1 files changed, 4 insertions(+), 6 deletions(-)

diff --git a/init/devices.c b/init/devices.c
index 1060dff..4faa38d 100644
--- a/init/devices.c
+++ b/init/devices.c
@@ -499,7 +499,8 @@ static void handle_device(const char *action, const char *devpath,
     int i;
 
     if(!strcmp(action, "add")) {
-        make_device(devpath, path, block, major, minor);
+        if (major >= 0 && minor >= 0)
+            make_device(devpath, path, block, major, minor);
         __system_property_set("ctl.dev_added",devpath);
         if (links) {
             for (i = 0; links[i]; i++)
@@ -513,7 +514,8 @@ static void handle_device(const char *action, const char *devpath,
                 remove_link(devpath, links[i]);
         }
         __system_property_set("ctl.dev_removed",devpath);
-        unlink(devpath);
+        if (major >= 0 && minor >= 0)
+            unlink(devpath);
     }
 
     if (links) {
@@ -537,10 +539,6 @@ static const char *parse_device_name(struct uevent *uevent, unsigned int len)
 {
     const char *name;
 
-    /* if it's not a /dev device, nothing else to do */
-    if((uevent->major < 0) || (uevent->minor < 0))
-        return NULL;
-
     /* do we have a name? */
     name = strrchr(uevent->path, '/');
     if(!name)
-- 
1.7.4.1

